scripts/src/javatests/org/oppia/android/scripts/maven/MavenDependenciesListWriterTest.kt

```diff
@@ -0,0 +1,308 @@
+import org.mockito.kotlin.mock
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.PrimaryLinkType
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+
+private const val DATA_BINDING_POM = "https://maven.google.com/androidx/databinding/" +
+  "databinding-adapters/3.4.2/databinding-adapters-3.4.2.pom"
+private const val PROTO_LITE_POM = "https://repo1.maven.org/maven2/com/google/protobuf/" +
+  "protobuf-lite/3.0.0/protobuf-lite-3.0.0.pom"
+private const val IO_FABRIC_POM = "https://maven.google.com/io/fabric/sdk/android/" +
+  "fabric/1.4.7/fabric-1.4.7.pom"
+private const val GLIDE_ANNOTATIONS_POM = "https://repo1.maven.org/maven2/com/github/" +
+  "bumptech/glide/annotations/4.11.0/annotations-4.11.0.pom"
+private const val FIREBASE_ANALYTICS_POM = "https://maven.google.com/com/google/firebase/" +
+  "firebase-analytics/17.5.0/firebase-analytics-17.5.0.pom"
+
+/** Tests for [MavenDependenciesListWriter]. */
+class MavenDependenciesListWriterTest {
+
+  private val LICENSE_DETAILS_INCOMPLETE_FAILURE = "License details are not completed."
+  private val COMPLETE_LICENSE_DETAILS_MESSAGE = "Please complete all the details" +
+    "for the following licenses:"
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  private val mockNetworkAndBazelUtils by lazy { initializeMockNetworkAndBazelUtils() }
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("scripts", "assets")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  // 1. Some License requires manual work, script fails.
+  // 2. Some Dependency requires manual work, script fails.
+  // 3. Some License and dependency both requires manual work, script fails.
+  // 4. Textproto is complete, script passes.
+  // 5. Some dependency contains empty license list, script fails.
+  // 6. Incomplte manual work, script fails.
+  // 7. Dependencies contain invalid links, script fails.
+
+  @Test
+  fun testLicenseNeedManualWork_scriptFailsWithExceptionAndCallOut() {
+    val dependencies = listOf<DependencyName>(
+      DependencyName.DATA_BINDING,
+      DependencyName.FIREBASE_ANALYTICS
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencies)
+    val mavenInstallJson = tempFolder.newFile("scripts/assets/maven_install.json")
+    writeMavenInstallJson(mavenInstallJson)
+    val pbFile = File("scripts/assets/maven_dependencies.pb")
+    pbFile.getParentFile().mkdirs()
+    val textProtoFile = tempFolder.newFile("scripts/assets/maven_dependencies.textproto")
+    mavenDependencyList.writeTo(pbFile.outputStream())
```

Comment:
This also does not work and throws exception - 
```
There was 1 failure:
1) testLicenseNeedManualWork_scriptFailsWithExceptionAndCallOut(org.oppia.android.scripts.maven.MavenDependenciesListWriterTest)
java.io.FileNotFoundException: scripts/assets/maven_dependencies.pb (Read-only file system)
        at java.io.FileOutputStream.open0(Native Method)
        at java.io.FileOutputStream.open(FileOutputStream.java:270)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
        at org.oppia.android.scripts.maven.MavenDependenciesListWriterTest.testLicenseNeedManualWork_scriptFailsWithExceptionAndCallOut(MavenDependenciesListWriterTest.kt:79)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

```

Reply:
However, we can probably make that work by passing the relative path of the binary file as a parameter. Because then we'll create this file inside the tempFolder and then it will not throw an exception.

Reply:
Already tracking this above.